File: Math/Python/sum_of_digits_until_single_digit.py
def sum_of_digits(n):
    while n >= 10:
        n = sum(map(int, str(n)))
    return n

print("Result:", sum_of_digits(9875))

# 9875 → 9+8+7+5=29 → 2+9=11 → 1+1=2

File: Math/Java/SumOfDigitsUntilSingleDigit.java
public class SumOfDigitsUntilSingleDigit {
    public static void main(String[] args) {
        int num = 9875;
        while (num >= 10) {
            int sum = 0;
            while (num > 0) {
                sum += num % 10;
                num /= 10;
            }
            num = sum;
        }
        System.out.println("Result: " + num);
    }
}